aea/crypto/helpers.py

```diff
@@ -87,58 +91,14 @@ def create_private_key(ledger_id: str, private_key_file: Optional[str] = None) -
     crypto.dump(open(private_key_file, "wb"))
 
 
-# TODO replace the If-This-Then-That paradigm
-def _try_generate_testnet_wealth(identifier: str, address: str) -> None:
+def try_generate_testnet_wealth(identifier: str, address: str) -> None:
     """
-    Generate wealth on a testnet.
+    Try generate wealth on a testnet.
 
     :param identifier: the identifier of the ledger
     :param address: the address to check for
     :return: None
     """
-    try:
-
-        if identifier == FETCHAI:
-            payload = json.dumps({"address": address})
-            response = requests.post(FETCHAI_TESTNET_FAUCET_URL, data=payload)
-            if response.status_code // 100 == 5:
-                logger.error("Response: {}".format(response.status_code))
-            else:
-                response_dict = json.loads(response.text)
-                if response_dict.get("error_message") is not None:
-                    logger.warning(
-                        "Response: {}\nMessage: {}".format(
-                            response.status_code, response_dict.get("error_message")
-                        )
-                    )
-                else:
-                    logger.info(
-                        "Response: {}\nMessage: {}\nDigest: {}".format(
-                            response.status_code,
-                            response_dict.get("message"),
-                            response_dict.get("digest"),
-                        )
-                    )  # pragma: no cover
-        elif identifier == ETHEREUM:
-            response = requests.get(ETHEREUM_TESTNET_FAUCET_URL + address)
-            if response.status_code // 100 == 5:
-                logger.error("Response: {}".format(response.status_code))
-            elif response.status_code // 100 in [3, 4]:
-                response_dict = json.loads(response.text)
-                logger.warning(
-                    "Response: {}\nMessage: {}".format(
-                        response.status_code, response_dict.get("message")
-                    )
-                )
-            elif response.status_code // 100 == 2:
-                response_dict = json.loads(response.text)
-                logger.info(
-                    "Response: {}\nMessage: {}".format(
-                        response.status_code, response_dict.get("message")
-                    )
-                )  # pragma: no cover
-    except Exception as e:
-        logger.warning(
-            "An error occured while attempting to generate wealth:\n{}".format(e)
-        )
-        sys.exit(1)
+    faucet_api = IDENTIFIER_TO_FAUCET_APIS.get(identifier, None)
```

Comment:
@marcofavorito thanks to common interface now this is generic...we can also make it user extendable in the future